packages/roosterjs-content-model-dom/lib/modelToDom/optimizers/optimize.ts

```diff
@@ -1,22 +1,38 @@
+import { enumerateInheritedStyle, removeUnnecessaryStyle } from './removeUnnecessaryStyle';
 import { isEntityElement } from '../../domUtils/entityUtils';
 import { mergeNode } from './mergeNode';
+import { removeUnnecessaryAttribute } from './removeUnnecessaryAttribute';
 import { removeUnnecessarySpan } from './removeUnnecessarySpan';
 
 /**
  * @internal
  */
-export function optimize(root: Node) {
+export function optimize(root: Node, isRecursive: boolean = false) {
     /**
      * Do no do any optimization to entity
      */
     if (isEntityElement(root)) {
         return;
     }
+    if (!isRecursive && root.parentElement != null) {
```

Comment:
Actually, instead of adding those attributes then remove them, it is better not to add them at all.

We can do this when generate HTML element, which is earlier than here. So no need to remove them here.

Another concern is perf. This function will be run to a lot of elements, so we need to be careful here, not adding too much loops.

If your requirement is about LI element, we can do this optimization when create LI in handleListItem.ts by adding those inline styles into context styles so styles with the same value will not be generated at all when create texts.

Let me see if I can make a PR.